src/shared/components/ncIconPicker/NcIconPicker.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+	<NcPopover container="body" popup-role="dialog" :shown.sync="showIconPicker">
+		<template #trigger="slotProps">
+			<slot v-bind="slotProps" />
+		</template>
+
+		<IconPicker :icons="icons" @select="select" />
+	</NcPopover>
+</template>
+
+<script>
+import iconData from '../../../../img/material/meta.json'
+
+import { NcPopover } from '@nextcloud/vue'
+import IconPicker from './partials/IconPicker.vue'
+
+export default {
+	name: 'NcIconPicker',
+
+	components: {
+		NcPopover,
+		IconPicker,
+	},
+
+	props: {
+		closeOnSelect: {
```

Comment:
Probably still on your list, just to not forget: would be good to pass in the current icon for the edit modal here as well

Reply:
I seen this is handled differently now, fixed it to properly pick up the currently set icon in my latest commit